pkg/certificate/manager.go

```diff
@@ -139,12 +139,12 @@ func (m *Manager) GetTrustDomain() string {
 // ShouldRotate determines whether a certificate should be rotated.
 func (m *Manager) ShouldRotate(c *Certificate) bool {
 	// The certificate is going to expire at a timestamp T
-	// We want to renew earlier. How much earlier is defined in renewBeforeCertExpires.
+	// We want to renew earlier. How much earlier is defined as a third of the certificate's validity duration.
 	// We add a few seconds noise to the early renew period so that certificates that may have been
 	// created at the same time are not renewed at the exact same time.
 	intNoise := rand.Intn(noiseSeconds) // #nosec G404
 	secondsNoise := time.Duration(intNoise) * time.Second
-	renewBefore := RenewBeforeCertExpires + secondsNoise
+	renewBefore := m.getValidityDurationForCertType(c.certType)/3 + secondsNoise
```

Comment:
unfortunately this will break if a user has changed their validity duration. Imagine a time of 24 hours, but then a user changes to 1 week.

Now we have expired certs.

A good heuristic might be this + a floor. So we can say 1/3 or 5 mins (whatever we set here, we should add a minimum accepted validity duration of 2x the renewal period, so 10 mins if 5 mins, or set 5 mins and 2.5 mins).

There's other ways to accomodate this as well, but this seems like a good first step to me